android-components/components/feature/prompts/src/main/java/mozilla/components/feature/prompts/identitycredential/ListItem.kt

```diff
@@ -0,0 +1,109 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.prompts.identitycredential
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.material.Text
+import androidx.compose.material.ripple.rememberRipple
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import mozilla.components.ui.colors.PhotonColors
+
+/**
+ * List item used to display an IdentityCredentials item that supports clicks
+ *
+ * @param title the Title of the item
+ * @param description the Description of the item
+ * @param modifier The modifier to apply to this layout.
+ * @param onClick Callback to handle when item is clicked.
+ * @param startComposable a Composable positioned at the start of the ListItem
+ *
+ */
+@Composable
+internal fun ListItem(
+    title: String,
+    description: String,
+    modifier: Modifier = Modifier,
+    onClick: () -> Unit,
+    startComposable: (@Composable () -> Unit)? = null,
+) {
+    // Used to propagate the ripple effect to the whole row
+    val interactionSource = remember { MutableInteractionSource() }
```

Comment:
For some reason I don't understand, not using this interactionSource gives a "solid" selected state that isn't a ripple :

https://github.com/mozilla-mobile/firefox-android/assets/460498/9057d4b3-8a6a-4b23-b5fb-e3e82d727565

This is the result I get when using the interactionSource:

https://github.com/mozilla-mobile/firefox-android/assets/460498/350fc643-18c5-402e-ac56-bc0227605e00

Reply:
If I had to guess, it's likely because of the lack of MaterialTheme wrapping all of the Compose code, which might be addressed by the patch we've talked about offline.